### ISSUE TITLE
Add the Lone Operative event.

### ISSUE THREAD

**Issue by PetMudstone** [unitystation/unitystation#7035]
### Description
The Lone Operative event basically just respawns a ghost as a lone Nuclear Operative, hence the name. The Lone Op spawns at an external spawnpoint (same one used for carp migration event) and has the objective of using the nuke disk to blow up the station. Unlike normal nuclear operatives, they are supposed to blow the station up using the station's on-board nuke.

**Comment by JukeJukestar** [comment]
I see some issues with this considering they start in space. Since people often start afk and im unsure if its possible to spawn someone with internals on, he would probably suffocate. A possible alternative would be spawning him in the syndicate lavaland base maybe? Just spawning him in maint seems a bit OP.

Also, he would need his own set of equipment instead of just having the standart Nukie spawn stuff, since Im pretty sure they start without a big oxy tank, no mags, no ammo, no nades and only one smg. Also consider a toolbelt with a fire extinguisher and an emag.

Should also be an admin spawn only event so it doesnt happen at a bad time. Other than that sounds like a good mid-round event to bring some life into the shift and keep the crew on their toes even after killing all the traitors

**Comment by Fogapod** [comment]
> respawns a ghost

this needs a system for ghost respawns before this is implemented. respawning afk people or people who do not want this is a bad idea.
in goonstation all ghosts get a notification when event happens and have a minute to click "yes" in order to be candidates for being respawned. this is different from ghost roles

**Comment by PerfectTangent** [comment]
> this needs a system for ghost respawns before this is implemented. respawning afk people or people who do not want this is a bad idea.
> in goonstation all ghosts get a notification when event happens and have a minute to click "yes" in order to be candidates for being respawned. this is different from ghost roles

We have a system for it. Noooney added one 7 months ago in #5675.

**Comment by Fogapod** [comment]
> We have a system for it. Noooney added one 7 months ago in #5675.

If you're talking about ghost role system where users have to select role from list (ancient engineer, for example), this is a bad idea. It favors users with lower ping and higher reaction speed who click on role faster instead of being fair and random.

**Comment by wraith-54321** [comment]
no, I think we have a version that is RNG based with a timer like in TG, its just not used

**Comment by PerfectTangent** [comment]
> > We have a system for it. Noooney added one 7 months ago in #5675.
> 
> If you're talking about ghost role system where users have to select role from list (ancient engineer, for example), this is a bad idea. It favors users with lower ping and higher reaction speed who click on role faster instead of being fair and random.

It has a buffer built in to prevent this problem.

**Comment by PerfectTangent** [comment]
or, at least, _some_ of the roles in the ghost role system have the buffer, like death squad, and that can be easily applied to this and other events. It's a surmountable problem.

**Comment by NoooneyDude** [comment]
> > We have a system for it. Noooney added one 7 months ago in #5675.
> 
> If you're talking about ghost role system where users have to select role from list (ancient engineer, for example), this is a bad idea. It favors users with lower ping and higher reaction speed who click on role faster instead of being fair and random.

This isn't strictly true.

All requests within a certain time initial time are added to a pool. Players are then randomly selected from this pool after the warmup duration expires. So, those who are really interested still get first choice, but shouldn't be limited by time-to-request.

However, the warmup period might need tweaking. I think it is set to one second?

Edit: ahh, Jack already explained as such...
